src/Microsoft.ML.Ensemble/Trainer/EnsembleTrainerBase.cs

```diff
@@ -54,27 +54,22 @@ public abstract class ArgumentsBase : LearnerInputBaseWithLabel
             public bool ShowMetrics;
 
             [Argument(ArgumentType.Multiple, HelpText = "Base predictor type", ShortName = "bp,basePredictorTypes", SortOrder = 1, Visibility = ArgumentAttribute.VisibilityType.CmdLineOnly)]
-            public SubComponent<ITrainer<RoleMappedData, IPredictorProducing<TOutput>>, TSig>[] BasePredictors;
+            public SubComponent<ITrainer<IPredictorProducing<TOutput>>, TSig>[] BasePredictors;
         }
 
         private const int DefaultNumModels = 50;
         /// <summary> Command-line arguments </summary>
-        protected readonly ArgumentsBase Args;
-        protected readonly int NumModels;
+        protected internal readonly ArgumentsBase Args;
+        protected internal readonly int NumModels;
 
         /// <summary> Ensemble members </summary>
-        protected readonly ITrainer<RoleMappedData, IPredictorProducing<TOutput>>[] Trainers;
+        protected internal readonly ITrainer<IPredictorProducing<TOutput>>[] Trainers;
 
         private readonly ISubsetSelector _subsetSelector;
-        protected ISubModelSelector<TOutput> SubModelSelector;
-        protected IOutputCombiner<TOutput> Combiner;
+        protected internal ISubModelSelector<TOutput> SubModelSelector;
+        protected internal IOutputCombiner<TOutput> Combiner;
 
-        protected List<FeatureSubsetModel<IPredictorProducing<TOutput>>> Models;
-
-        private readonly bool _needNorm;
-        private readonly bool _needCalibration;
-
-        internal EnsembleTrainerBase(ArgumentsBase args, IHostEnvironment env, string name)
+        protected internal EnsembleTrainerBase(ArgumentsBase args, IHostEnvironment env, string name)
```

Comment:
Since this class is `abstract`, there is no point in making this `protected internal`.  `protected internal` means it can be called from either a subclass, or any other class in this current assembly (for friend assembly).  Since it is `abstract`, it can only be called from a sub-class - no other code can directly instantiate this class.  So instead, I think this should just be marked `protected`.